File: 22Jan2024.py
# Jai Shree Ram
# Problem Link :- https://leetcode.com/problems/set-mismatch/

from ast import List


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        st = set()
        ans= [0]*2
        for num in nums :
            if(num in st) :
                ans[0]=num
            st.add(num)
        for i in range(1,len(nums)+1) :
            if(i not in st) :
                ans[1]=i
        return ans

File: 22Jan2024.java
// Jai Shree Ram
// Problem Link :- https://leetcode.com/problems/set-mismatch/

import java.util.HashSet;

class Solution {
    public int[] findErrorNums(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int[] ans = new int[2];
        for (int num : nums) {
            if (set.contains(num)) {
                ans[0] = num;
            }
            set.add(num);
        }

        for (int i = 1; i <= nums.length; i++) {
            if (!set.contains(i))
                ans[1] = i;
        }
        return ans;
    }
}